pylxd/models/instance.py

```diff
@@ -224,6 +253,38 @@ def recursive_put(self, src, dst, mode=None, uid=None, gid=None):
                         if response.status_code != 200:
                             raise LXDAPIException(response)
 
+        def recursive_get(self, remote_path, local_path):
+            """Recursively pulls a directory from the container.
+            Pulls the directory named `remote_path` from the container and
+            creates a local folder named `local_path` with the
+            content of `remote_path`.
+            If `remote_path` is a file, it will be copied to `local_path`.
+            :param remote_path: The directory path on the container.
+            :type remote_path: str
+            :param local_path: The path at which the directory will be stored.
+            :type local_path: str
+            :return:
+            :raises: LXDAPIException if an error occurs
+            """
+            response = self._endpoint.get(
+                params={'path': remote_path}, is_api=False)
+
+            print(response)
+            if "X-LXD-type" in response.headers:
+                print("1")
+                if response.headers["X-LXD-type"] == "directory":
+                    print("2")
+                    os.mkdir(local_path)
```

Comment:
What's the story on file permissions here.  At the moment they will be what ever is the umask for the parent dir?  Should they be copied/set from what is coming from the server?  If that's not the use case, then maybe put a note in the function header.

Reply:
Thank you very much for your feedback and your thoughts! Honestly, I did not think about the possibility to set the umask to the information coming from the server. Do you think that would be the better way to go? Is there an easy way to get these information from the server that you are thinking of?

Reply:
> Thank you very much for your feedback and your thoughts! Honestly, I did not think about the possibility to set the umask to the information coming from the server. Do you think that would be the better way to go? Is there an easy way to get these information from the server that you are thinking of?

Well, the REST api at https://github.com/lxc/lxd/blob/master/doc/rest-api.md#10instancesnamefiles appears to indicate that the perms are available for the files and directories that are pulled from the LXD instance; I'm guessing that they could be used to set the permissions.  Although, there is a uid and gid, I doubt that it makes sense to use them, though, as they wouldn't necessarily translate to the host machine in any meaningful way.

I guess my question was about "is this useful in the scope of the use-case that is driving the patch"?  If not, then just perhaps add an 'TODO' comment saying something like "we considered using the X-LXD-uid, X-LXD-gid, and X-LXD-mode header information, but it was beyond the scope of this change."?